tests/test_preview.py

```diff
@@ -0,0 +1,271 @@
+
+
+def test_model_preview_properties(test_dir, wlmutils):
+    """
+    Test correct preview output properties for Model preview
+    """
+    # Prepare entities
+    exp_name = "test_model_preview_parameters"
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+    rs1 = exp.create_run_settings("echo", ["hello", "world"])
+    rs2 = exp.create_run_settings("echo", ["spam", "eggs"])
+    hello_world_model = exp.create_model("echo-hello", run_settings=rs1)
+    spam_eggs_model = exp.create_model("echo-spam", run_settings=rs2)
+    preview_manifest = Manifest(hello_world_model, spam_eggs_model)
+
+    # Execute preview method
+    rendered_preview = previewrenderer.render(exp, preview_manifest)
+
+    # Evaluate output for hello world model
+    assert "echo-hello" in rendered_preview
+    assert "/usr/bin/echo" in rendered_preview
+    assert "hello" in rendered_preview
+    assert "world" in rendered_preview
+    assert "echo-hello" == hello_world_model.name
+    assert "/usr/bin/echo" == hello_world_model.run_settings.exe[0]
+    assert "hello" == hello_world_model.run_settings.exe_args[0]
+    assert "world" == hello_world_model.run_settings.exe_args[1]
+    # Evaluate outputfor spam eggs model
+    assert "echo-spam" in rendered_preview
+    assert "/usr/bin/echo" in rendered_preview
+    assert "spam" in rendered_preview
+    assert "eggs" in rendered_preview
+    assert "echo-spam" == spam_eggs_model.name
+    assert "/usr/bin/echo" == spam_eggs_model.run_settings.exe[0]
+    assert "spam" == spam_eggs_model.run_settings.exe_args[0]
+    assert "eggs" == spam_eggs_model.run_settings.exe_args[1]
+
+
+def test_model_with_tagged_files(fileutils, test_dir, wlmutils):
+    """
+    Test multiple models and tagged files for Model configuration in preview.
+    """
+    # Prepare entities
+    exp_name = "test_model_preview_parameters"
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+
+    model_params = {"port": 6379, "password": "unbreakable_password"}
+    model_settings = RunSettings("bash", "multi_tags_template.sh")
+    rs2 = exp.create_run_settings("echo", ["spam", "eggs"])
+
+    hello_world_model = exp.create_model(
+        "echo-hello", run_settings=model_settings, params=model_params
+    )
+    spam_eggs_model = exp.create_model("echo-spam", run_settings=rs2)
+
+    config = fileutils.get_test_conf_path(
+        osp.join("generator_files", "multi_tags_template.sh")
+    )
+    hello_world_model.attach_generator_files(to_configure=[config])
+    exp.generate(hello_world_model, overwrite=True)
+
+    preview_manifest = Manifest(hello_world_model, spam_eggs_model)
+
+    # Execute preview method
+    rendered_preview = previewrenderer.render(exp, preview_manifest)
+
+    # Evaluate output for hello_world_model
+    assert "Model name" in rendered_preview
```

Comment:
Consider separate tests or renaming test.
- test named "test w/tagged files" but also validates model params, batch args, password, etc
- split test so asserts for model params don't cloud tests for outputs related to files

Reply:
separated tests